client/app/src/main/java/com/example/criengine/Profile.java

```diff
@@ -1,36 +1,47 @@
 package com.example.criengine;
 
-import java.util.Map;
+import java.util.ArrayList;
 
 public class Profile {
 
+    private String userID;
+    private String email;
     private String username;
     private String phone;
     private String firstname;
     private String lastname;
-    private String email;
     private String bio;
+    private ArrayList<String> booksOwned;
+    private ArrayList<String> booksBorrowedOrRequested;
+
+
+    public Profile() {
+        this.booksOwned = new ArrayList<>();
+        this.booksBorrowedOrRequested = new ArrayList<>();
+    }
 
-    public Profile(String username, String phone, String firstname, String lastname, String email) {
+    public Profile(String userID, String email, String username, String phone, String firstname, String lastname, String bio) {
+        this.userID = userID;
+        this.email = email;
         this.username = username;
         this.phone = phone;
         this.firstname = firstname;
         this.lastname = lastname;
-        this.email = email;
-    }
-
-    public Profile() {
-
+        this.bio = bio;
+        this.booksOwned = new ArrayList<>();
+        this.booksBorrowedOrRequested = new ArrayList<>();
     }
 
-    public Profile(Map<String, Object> map) {
-        this.username = (String) map.get("username");
-        this.phone = (String) map.get("phone");
-        this.firstname = (String) map.get("firstname");
-        this.lastname = (String) map.get("lastname");
-        this.email = (String) map.get("email");
-        this.bio = (String) map.get("bio");
-
+    public Profile(String userID, String email, String username, String phone, String firstname, String lastname) {
+        this.userID = userID;
+        this.email = email;
+        this.username = username;
+        this.phone = phone;
+        this.firstname = firstname;
+        this.lastname = lastname;
+        this.bio = null;
+        this.booksOwned = new ArrayList<>();
+        this.booksBorrowedOrRequested = new ArrayList<>();
```

Comment:
these two should include the type String to match the declaration (probably should be a list of book objects though)

Reply:
the string is implicit when its being set to string.
probably wont make it book objects since that would be deep nesting, but its possible the local profile object should be able to handle that. I'll look into it

Reply:
the problem I see with that is what the book objects would be stored in the database as, but I suppose we could store them as references.